lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1303,6 +1307,45 @@ getFeePerByteFromWalletPParams pp =
   where
     LinearFee LinearFunction{slope} = getFeePolicy $ txParameters pp
 
+-- | Like the 'TxConstraints' field 'txRewardWithdrawalCost', but with added
+-- support for shared wallets via the 'CA.ScriptTemplate' argument.
+--
+-- We may or may not want to support shared wallets in the full txConstraints.
+_txRewardWithdrawalCost
+    :: Write.FeePerByte
+    -> Either CA.ScriptTemplate TxWitnessTag
+    -> Coin
+    -> Coin
+_txRewardWithdrawalCost feePerByte witType =
+    toWallet
+    . Write.feeOfBytes feePerByte
+    . unTxSize
+    . _txRewardWithdrawalSize witType
+
+-- | Like the 'TxConstraints' field 'txRewardWithdrawalSize', but with added
+-- support for shared wallets via the 'CA.ScriptTemplate' argument.
+--
+-- We may or may not want to support shared wallets in the full txConstraints.
+_txRewardWithdrawalSize
+    :: Either CA.ScriptTemplate TxWitnessTag
+    -> Coin
+    -> TxSize
+_txRewardWithdrawalSize _ (Coin 0) = TxSize 0
+_txRewardWithdrawalSize witType _ = TxSize
+        $ fromMaybe (error "txRewardWithdrawalSize: negative size")
+        $ intCastMaybe
+        $ sizeOf_Withdrawals 1 - sizeOf_Withdrawals 0 + wits
+      where
+        wits = case witType of
```

Comment:
```suggestion
        witSize :: TxSize
        witSize = case witType of
```

Reply:
Going with `witSize` without `TxSize` for now in this PR.